doc/source/Tutorials/GettingStarted/index.md

```diff
@@ -1,407 +1,499 @@
-[2]: ../../Reference/NodeReference/GafferScene/SceneReader.md
-[3]: ../../Reference/UIReference/GraphEditor.md
-[4]: ../../Reference/UIReference/Viewer.md
+
+## Pinning an Editor to a Node ##
+
+As mentioned earlier, the _Viewer_, _Scene Hierarchy_, and _Node Editor_ (each an **editor**) show their respective outputs of the currently selected node. This is not always convenient, because often you will need to edit one node while viewing the output of another. You can solve this by **pinning** an editor while a node is selected, which keeps that editor focused on the node.
+
+To make switching between viewing Gaffy's geometry and the render easier, you can modify the UI so you're working with two Viewers. First, start by pinning the last node in the graph that contains the scene:
+
+1. Select the InteractiveAppleseedRender node.
+
+2. Click ![the pin button](images/targetNodesUnlocked.png "Pin button") at the top-right
+  of the top pane. The pin button will highlight: ![highlighted pin](images/targetNodesLocked.png "Highlighted pin").
+
+The _Viewer_ is now locked to the InteractiveAppleseedRender node's scene (which contains all of the parts from its upstream scenes), and will only show that scene, even if you deselect it or select a different node.
+
+Next, pin the same node to the _Scene Hierarchy_. This time, use the middle-click shortcut:
+
+1. Select the InteractiveAppleseedRender node.
+
+2. Middle-click and drag the InteractiveAppleseedRender node from the _Graph Editor_ into the
+  _Scene Hierarchy_.
+
+As with the _Viewer_, the _Scene Hierarchy_ will now remain locked to the output of InteractiveAppleseedRender, regardless of your selection. Now you are free to select any node for editing in the _Node Editor_, but you will always be viewing the final results of the last node in the graph.
+
+For the final adjustment to the UI, create another _Viewer_ in the top-left panel, and pin the Catalogue node to it:
+
+1. At the top-right of the top panel, click ![the layout menu button](images/layoutButton.png "Layout menu button") to open the layout menu.
+
+2. Select _Viewer_. A new _Viewer_ will appear on the panel next to the first one.
+
+3. Middle-click and drag the Catalogue node onto the new _Viewer_. That _Viewer_ is now pinned to that node.
+
+Now you can switch between the scene's geometry (first _Viewer_) and the rendered image (second _Viewer_). Now it is time to shade Gaffy.
+
+
+## Adding Shaders and Lighting ##
+
+It's time to add shaders and lighting. It will help to think of a graph as being composed of three distinct phases: generating the geometry, applying the lighting and shading, and rendering. Lights act like another scene object, so you can group one into the scene like the other nodes. The shaders, however, are different: in order for the render option nodes to apply to the geometry and to inherit the shaders, you will need to add them somewhere between the scene-creating nodes and render options nodes.
```

Comment:
This paragraph is a bit funny perhaps. Shaders aren't **options**, they're **attributes**. And Lights aren't **objects**, they're **attributes** (shaders). While you can group lights into the scene via the various Light nodes, you also can assign them to existing locations (thought I don't know if that's easily exposed to users). Also, the order of nodes doesn't matter here (render options can be specified before and/or after attributes). Maybe @danieldresser can have a go at rewording it?

Reply:
Will clarify and correct.

Reply:
We also want to discourage people from doing two separate phases, first generating all the geometry and then applying all shaders - this is what often happens in the IE pipe, and it's really inefficient.  It's better to have lots of small branches feeding into your graph, each of which loads some geometry and sets up shaders on just that geo.

Here are some thing we could say: "Lights are created at their own location, and can be added anywhere in the graph.  Shaders are added to geometry, so shader assignment need to be downstream from the shaders they target.  Render options can be applied anywhere in the graph, but because they are global, and nodes like Group and Parent only pass through options from their first input, options are usually applied further down the graph, after all geo has been combined into one stream."

Not sure all of that makes sense to talk about here.

Reply:
Probably not, but that helps my understanding, and I'll include the bit about applying shaders ASAP (good reinforcement for the later complexity articles).